test/extended/builds/s2i_extended_build.go

```diff
@@ -0,0 +1,140 @@
+package builds
+
+import (
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[builds][Slow] s2i extended build", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc                 = exutil.NewCLI("extended-build", exutil.KubeConfigPath())
+		testDataDir        = exutil.FixturePath("testdata", "build-extended")
+		scriptsFromRepoBc  = filepath.Join(testDataDir, "bc-scripts-in-repo.json")
+		scriptsFromUrlBc   = filepath.Join(testDataDir, "bc-scripts-by-url.json")
+		scriptsFromImageBc = filepath.Join(testDataDir, "bc-scripts-in-the-image.json")
+	)
+
+	g.Describe("with scripts from the source repository", func() {
+		oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+		g.It("should use assemble-runtime script from the source repository", func() {
+			const buildConfigName = "java-extended-build-from-repo"
+			const buildName = buildConfigName + "-1"
+
+			g.By("creating build config")
+			err := oc.Run("create").Args("-f", scriptsFromRepoBc).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("running the build")
+			err = oc.Run("start-build").Args(buildConfigName).Execute()
```

Comment:
Ok, I added. But I decided to output this logs only when `start-build` fails and not always as in the example.